### DISCUSSION TITLE
[Source Code] Conditionally load allowed providers only

### DISCUSSION THREAD

**Discussion by kvj** [music-assistant/support#2036] [Q&A]
Hey, I'm looking at the code [here](https://github.com/music-assistant/server/blob/4938443c4feacc76d8ac28c0b233410bed22c50d/music_assistant/server/server.py#L571) and it looks like that all available providers are always loaded. Do you think it's possible to add (or accept contribution) a configuration key to only load allowed providers? I don't need ~ half of them and see no reason to install the required dependencies for them. 

As I only installed required python dependencies for providers I need, now mass shows quite a bunch of errors (missing libraries) at the startup, but works as expected

PS: I understand that mass code structure is inspired by the HA code, but HA only loads (and correctly install dependencies) enabled/configured components and not all available

**Comment by marcelveldt** [comment]
You are reading it wrong. Only the provider manifests are loaded.
You are supposed to run the server using one of the SUPPORTED installation methods, being either the docker container or the HA add-on. I don't support manual installs, sorry.

**Reply by kvj** [reply]
Ok I get your point. I will support venv installation by myself, as none of the SUPPORTED installation methods fit my case.

BTW, [here](https://github.com/music-assistant/server/blob/443bb83ff0bc22bd31e5c968197983e1b7d4bed8/music_assistant/server/helpers/util.py#L33) you could use `sys.executable` instead of `"python3"` to support more environments
